poshc2/client/Help.py

```diff
@@ -566,55 +556,54 @@
-            'invoke-psexecpayload', 'invoke-wmijspayload', 'invoke-wmipayload', 'invoke-dcompayload',
-            "get-ipconfig", 'invoke-wmijspbindpayload', 'get-lapspasswords', "get-inveigh", "runas-netonly", "invoke-edrchecker",
-            "get-powerstatus", "loadpowerstatus", "get-injectedthread", "enable-rotation", "get-rotation", "invoke-urlcheck", "searchhistory"]
-
-# post help commands python implant
-UXCOMMANDS = ["label-implant", "remove-label", "unhide-implant", "hide-implant", "help", "searchhelp", "python", "loadmodule",
-              "loadmoduleforce", "get-keystrokes", "back", "upload-file", "download-file", "install-persistence", "remove-persistence", "sai",
-              "startanotherimplant-keepfile", "get-screenshot", "startanotherimplant", "pwd", "id", "ps", "beacon", "kill-implant", "linuxprivchecker", "quit", "searchhistory"]
-
-# post help commands sharp implant
-SHARPCOMMANDS = ["get-userinfo", "get-idletime", "stop-keystrokes", "start-keystrokes", "start-keystrokes-writefile", "get-keystrokes", "move", "label-implant", "remove-label", "upload-file", "quit",
-                 "download-file", "get-content", "ls-recurse", "turtle", "cred-popper", "resolveip", "resolvednsname", "testadcredential",
-                 "testlocalcredential", "get-screenshot", "modulesloaded", "get-serviceperms", "unhide-implant", "arpscan", "ls", "pwd", "dir",
-                 "inject-shellcode", "start-process", "start-shortcut", "run-exe", "run-dll", "hide-implant", "help", "searchhelp", "listmodules", "loadmodule",
-                 "loadmoduleforce", "back", "ps", "beacon", "kill-implant", "get-screenshotmulti", "safetydump", "seatbelt", "sharpup",
-                 "sharphound", "rubeus", "sharpview", "kill-process", "watson", "get-hash", "migrate", "sharpsocks", "safetykatz", "get-computerinfo",
-                 "get-dodgyprocesses", "sharpweb", "bypass-amsi", "sharpsc", "dcomexec", "smbexec", "wmiexec", "sharpwmi", "sharpcookiemonster", "stop-screenshotmulti",
-                 "get-screenshotallwindows", "del", "kill-process", "posh-delete", "getremoteprocesslisting", "getremoteprocesslistingall", "sslinspectioncheck",
-                 "dynamic-code", "startdaisy", "mimikatz", "searchhistory", "stopdaisy", "portscan", "sharpdpapi", "sharpchrome", "sweetpotato",
-                 "pbind-connect", "pbind-pslo", "pbind-loadmodule", "pbind-command", "pbind-kill", "run-exe-background", "run-dll-background",
-                 "inveigh", "stopinveigh", "enable-rotation", "get-rotation", "loadpowerstatus", "getpowerstatus", "stoppowerstatus"]
+
+PRECOMMANDS = []
+COMMANDS = []
+UXCOMMANDS = []
+SHARPCOMMANDS = []
+
+special_characters = "!@#$%^&*()+=."
+
+#pre_help
+for line in pre_help.splitlines():
+	try:
+		entry = line.split(None, 1)[0]
+		if entry not in PRECOMMANDS and not any(char in special_characters for char in entry):
+			PRECOMMANDS.append(entry)
+	except IndexError:
+		pass
```

Comment:
As we're accessing the first index here the only time we'll get an IndexError is if the line is empty.
A better way of doing this then would be to check if the line is empty and if not add it, as opposed to trying to access something that may not exist and catching the error.

E.g. 

```
for line in pre_help.splitlines():
    try:
        line = line.strip()
        if line:
            entry = line.split(None, 1)[0]
            ...
    except Exception:
        print_bad("Error building prehelp")
        traceback.print_exc()
```

Without the except IndexError